src/Microsoft.VisualStudio.Editors/PropPages/PackagePropPage.vb

```diff
@@ -230,7 +285,117 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
+                LicenseFileNameTextBox.Text = ""
+            Else
+                PackageLicenseExpression.Text = ""
+            End If
+        End Sub
+
+        Private Sub LicenseTypeFirstSelected()
+            'When the project has neither of the new license properties AND it has the license URL property and a new license type is selected
+            If (_licenseUrlDetected) Then
+                LicenseLineLabel.BackColor = Drawing.SystemColors.ControlDark
+                LicenseLineLabel.Size = New Drawing.Size(LicenseLineLabel.Size.Width, 1)
+                SetCommonPropertyValue(GetPropertyDescriptor("PackageLicenseUrl"), "")
+                _licenseUrlDetected = False
+            End If
+        End Sub
+
+        Private Sub LicenseExpressionRadioButton_CheckedChanged(sender As Object, e As EventArgs) Handles LicenseExpressionRadioButton.CheckedChanged
+            If (Not _newLicensePropertyDetectedAtInit) Then
+                LicenseTypeFirstSelected()
+                _newLicensePropertyDetectedAtInit = False
+            End If
+            If (LicenseExpressionRadioButton.Checked) Then
+                SetLicenseRadioButtons(True)
+            End If
+        End Sub
+
+        Private Sub LicenseFileRadioButton_CheckChanged(sender As Object, e As EventArgs) Handles LicenseFileRadioButton.CheckedChanged
+            If (Not _newLicensePropertyDetectedAtInit) Then
+                LicenseTypeFirstSelected()
+                _newLicensePropertyDetectedAtInit = False
+            End If
+            If (LicenseFileRadioButton.Checked) Then
+                SetLicenseRadioButtons(False)
+            End If
+        End Sub
+
+        Private Sub PackageLicenseExpression_Changed(sender As Object, e As EventArgs) Handles PackageLicenseExpression.TextChanged
+            If (PackageLicenseExpression.Text IsNot "" And Not PackageLicenseExpression.Enabled) Then
+                'The license expression is not selected, and the text was changed while it was disabled
+                'This means there was probably an undo which populated the textbox with text, so give it back control
+                'I don't believe that undo will work with the license file text box because it is not user populated
+                SetLicenseRadioButtons(True)
+            End If
+        End Sub
+
+        'These GotFocus methods are for when the property page is first entered and neither have a value
+        'It would make sense to selected the corresponding radio button when the text box recieves focus
+        Private Sub PackageLicenseExpression_GotFocus(sender As Object, e As EventArgs) Handles PackageLicenseExpression.GotFocus
+            SetLicenseRadioButtons(True)
+        End Sub
+
+        Private Sub LicenseFileNameTextBox_GotFocus(sender As Object, e As EventArgs) Handles LicenseFileNameTextBox.GotFocus
+            SetLicenseRadioButtons(False)
```

Comment:
Are you intentionally not calling `LicenseTypeFirstSelected` here?

If not, you could move the four identical lines in the radio button change handlers into `SetLicenseRadioButtons` directly.

Reply:
Great catch! Fixed.